src/routes/CourseRoute.js

```diff
@@ -125,26 +130,40 @@ class CourseRoute extends React.Component {
   }
 
   render() {
-    const { handlers, resources, mutator } = this.props;
+    const { handlers, history, resources, mutator } = this.props;
+
+    const data = {
+      course: { ...get(resources, 'course.records[0]', {}) },
+      crossListed: get(resources, 'crossListed.records', []),
+      reserves: get(resources, 'reservesForCourse.records', []),
+      items: get(resources, 'items.records', []),
+    };
 
     return (
-      <Course
-        data={{
-          course: { ...get(resources, 'course.records[0]', {}) },
-          crossListed: get(resources, 'crossListed.records', []),
-          reserves: get(resources, 'reservesForCourse.records', []),
-          items: get(resources, 'items.records', []),
-        }}
-        handlers={{
-          ...handlers,
-          onClose: this.handleClose,
-          onCrosslist: this.handleCrosslist,
-          onEdit: this.handleEdit,
-        }}
-        isLoading={get(resources, 'course.isPending', true)}
-        resources={resources}
-        mutator={mutator}
-      />
+      <>
+        <Course
+          data={data}
+          handlers={{
+            ...handlers,
+            onClose: this.handleClose,
+            onCrosslist: this.handleCrosslist,
+            onDuplicate: () => this.setState({ showDuplicateCourseModal: true }),
+            onEdit: this.handleEdit,
+          }}
+          isLoading={get(resources, 'course.isPending', true)}
+          resources={resources}
+          mutator={mutator}
+        />
+        { // Conditionally render so that autofocus works naturally when mounting and state is reset
```

Comment:
I'd not come across this idiom before. It troubles me that we need to do this. Should we be wrapping _all_ modal inclusions like this now?

Reply:
Turns out we used to need it but we no longer do! Hooray for fixes from Rasmus and John. 😄

Reply:
Happy ending!